numba_dpex/experimental/core/types/kernel_api/items.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-FileCopyrightText: 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Defines numba types for Item and NdItem classes"""
+
+from numba.core import errors, types
+
+
+class ItemType(types.Type):
+    """Numba-dpex type corresponding to
+    :class:`numba_dpex.kernel_api.Item`
+    """
+
+    def __init__(self, ndim: int):
+        self._ndim = ndim
+        if ndim < 1 or ndim > 3:
+            raise errors.TypingError(
+                "ItemType can only have 1, 2 or 3 dimensions"
+            )
+        super().__init__(name="Item<" + str(ndim) + ">")
+
+    @property
+    def ndim(self):
+        """Returns number of dimensions"""
+        return self._ndim
+
+    @property
+    def key(self):
+        """Numba type specific overload"""
+        return self._ndim
+
+    def cast_python_value(self, args):
+        raise NotImplementedError
+
+
+class NdItemType(types.Type):
+    """Numba-dpex type corresponding to
```

Comment:
Docstring most likely will fit on one line < 80 chars

Reply:
Yea, it got shrink since last rebase